activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java

```diff
@@ -1545,8 +1545,9 @@ long updateIndex(Transaction tx, KahaAddMessageCommand command, Location locatio
         if (previous == null) {
             previous = sd.messageIdIndex.put(tx, command.getMessageId(), id);
             if (previous == null) {
-                incrementAndAddSizeToStoreStat(tx, command.getDestination(), location.getSize());
                 sd.orderIndex.put(tx, priority, id, new MessageKeys(command.getMessageId(), location));
```

Comment:
Shouldn't we use the `IndexLock` here to have the `orderIndex` and store stat updated atomically ? Just wondering.

Reply:
The entire method is already protected by the index write lock here: https://github.com/apache/activemq/blob/0b95f93449e58a855efd2fb03731432f396b18e1/activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java#L1342

This change is probably not even necessary since it's already under a lock and any exception should abort and shut things down normally but it just seemed to make more sense to not increment until adding to the index. but really it doesn't much matter since the index isn't commuted anyways if it isn't successful.

Reply:
Good point.

Reply:
Yeah your comment was good as it made me realize that change was pointless as I forgot the index isn't even committed until after the transaction completes. One could argue we could wait to update the counters until after transaction finishes but that would be a much larger change and tricky to guarantee things are in fact still atomic since you process it later outside of where you have the information you need and the lock. Plus the default behavior is to shutdown the broker on IoException anyways. The current behavior has been fine for many years so don't need to change it I don't think.